authorization/resource/resource_type_scope_blackbox_test.go

```diff
@@ -0,0 +1,154 @@
+	s.DBTestSuite.PopulateDBTestSuite(s.ctx)
+}
+
+func (s *resourceTypeScopeBlackBoxTest) SetupTest() {
+	s.DB.LogMode(true)
+	s.repo = resource.NewResourceTypeScopeRepository(s.DB)
+	s.resourceTypeRepo = resource.NewResourceTypeRepository(s.DB)
+	s.clean = cleaner.DeleteCreatedEntities(s.DB)
+}
+
+func (s *resourceTypeScopeBlackBoxTest) TearDownTest() {
+	s.clean()
+}
+
+func (s *resourceTypeScopeBlackBoxTest) TestOKToDelete() {
+	t := s.T()
+	res.Require(t, res.Database)
+
+	// create 2 resources types, where the first one would be deleted.
+	resourceTypeScope := createAndLoadResourceTypeScope(s)
+	createAndLoadResourceTypeScope(s)
+
+	err := s.repo.Delete(s.ctx, resourceTypeScope.ResourceTypeScopeID)
+	assert.Nil(s.T(), err)
+
+	// lets see how many are present.
+	resourceTypeScopes, err := s.repo.List(s.ctx, nil)
+	require.Nil(s.T(), err, "Could not list resource type scopes")
+	require.True(s.T(), len(resourceTypeScopes) > 0)
+
+	for _, data := range resourceTypeScopes {
+		// The resource type scope 'resourceTypeScope' was deleted and rest were not deleted, hence we check
+		// that none of the resource type scope objects returned include the one deleted.
+		require.NotEqual(s.T(), resourceTypeScope.ResourceTypeScopeID.String(), data.ResourceTypeScopeID.String())
+	}
+}
+
+func (s *resourceTypeScopeBlackBoxTest) TestOKToLoad() {
+	t := s.T()
+	res.Require(t, res.Database)
+
+	createAndLoadResourceTypeScope(s)
+}
+
+func (s *resourceTypeScopeBlackBoxTest) TestExistsResourceTypeScope() {
+	t := s.T()
+	res.Require(t, res.Database)
+
+	t.Run("resource type scope exists", func(t *testing.T) {
+		//t.Parallel()
+		resourceTypeScope := createAndLoadResourceTypeScope(s)
+		// when
+		_, err := s.repo.CheckExists(s.ctx, resourceTypeScope.ResourceTypeScopeID.String())
+		// then
+		require.Nil(t, err)
+	})
+
+	t.Run("resource type scope doesn't exist", func(t *testing.T) {
+		//t.Parallel()
+		// Check not existing
+		_, err := s.repo.CheckExists(s.ctx, uuid.NewV4().String())
+		// then
+		require.IsType(s.T(), errors.NotFoundError{}, err)
+	})
+}
+
+func (s *resourceTypeScopeBlackBoxTest) TestOKToSave() {
+	t := s.T()
+	res.Require(t, res.Database)
+
+	resourceTypeScope := createAndLoadResourceTypeScope(s)
+
+	resourceTypeScope.Name = "newResourceTypeScopeNameTestType"
+	err := s.repo.Save(s.ctx, resourceTypeScope)
+	require.Nil(s.T(), err, "Could not update resourceTypeScope")
+
+	updatedResourceTypeScope, err := s.repo.Load(s.ctx, resourceTypeScope.ResourceTypeScopeID)
+	require.Nil(s.T(), err, "Could not load resource type scope")
+	assert.Equal(s.T(), resourceTypeScope.Name, updatedResourceTypeScope.Name)
+	assert.Equal(s.T(), resourceTypeScope.Description, "Collaborators may perform many operations within an area")
+}
+
+func createAndLoadResourceTypeScope(s *resourceTypeScopeBlackBoxTest) *resource.ResourceTypeScope {
+
+	resourceType := &resource.ResourceType{
+		ResourceTypeID: uuid.NewV4(),
+		Name:           "Area" + uuid.NewV4().String(),
```

Comment:
Let's add the test name to all Names of all records created by tests. If the test doesn't clean the created records then it will be easy to identify what test is responsible for that. It's good to use in all DB tests.